src/platform/tests/TestPlatformMgr.cpp

```diff
@@ -55,14 +56,29 @@ static void TestPlatformMgr_InitShutdown(nlTestSuite * inSuite, void * inContext
 
 static void TestPlatformMgr_BasicEventLoopTask(nlTestSuite * inSuite, void * inContext)
 {
+    std::atomic<int> counterRun{ 0 };
+
     CHIP_ERROR err = PlatformMgr().InitChipStack();
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
 
-    err = PlatformMgr().StartEventLoopTask();
-    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    // Start/stop the event loop task a few times.
+    for (size_t i = 0; i < 3; i++)
+    {
+        err = PlatformMgr().StartEventLoopTask();
+        NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+        // Verify that the event loop will not exit until we tell it to by
+        // scheduling few lambdas (for the test to pass, the event loop will
+        // have to process more than one event).
+        DeviceLayer::SystemLayer().ScheduleLambda([&counterRun]() { counterRun++; });
+        chip::test_utils::SleepMillis(10);
```

Comment:
Can we avoid sleep in tests? Could we use some cond/barriers?

Reply:
To be honest I don't know how to implement that without a sleep (somewhere). The problem is that it seems (I have not analyzed the code in details) that `LayerImplSelect::HandleEvents()` handles all pending events at once. Without the sleep between schedule calls (I can implement that with a timer, but technically it will be a sleep anyway) I've got only one iteration of `do { ... } while()` in `_RunEventLoop()`...

Also, one more thing is not clear for me. Maybe someone will explain that. Why we've got `do {...} while` instead of `while(){...}`? If that loop would be implemented in the latter form, the bug would have been detected earlier, because this test runner starts and stops event loop task in few test cases and it works only because event loop always executes `HandleEvents()`.

Reply:
Lets add this explanation at the top of the sleep, to guide others to remove it if applicable.